schema-layer/data-structures/generics.md

```diff
@@ -0,0 +1,109 @@
+# Specification: IPLD Generics
+
+**Status: Prescriptive - Exploratory**
+
+Organizing IPLD data into usable, efficient, complex data structures spanning many blocks aimed for use by end-user applications.
+
+This document will re-use some terms found in the [IPLD data model](/IPLD-Data-Model-v1.md). 
+
+IPLD Generics offer codec agnostic programming interfaces for all common operations users can currently accomplish on Data-Model [“Kinds”]().
+
+Contents:
+
+  * [Motivation](#motivation)
+  * [Requirements](#requirements)
+  * [Operations](#operations)
+  * [Implementations](#implementations)
+
+## Motivation
+
+Even before the IPLD Data-Model was formally specified developers were creating multi-block data-structures with similar semantics to single-block primitives. The most illustrative example of this is the `dag-pb` HAMT implementation used by IPFS for large directories.
+
+These early implementations of multi-block data structures exposed several problems.
+
+  * They are lacking self-description. A consumer of a graph containing these structures would have to have logic on top of IPLD and vary the way it performs operations on there data structures.
+  * Implementations of these data structures cannot perform operations on each other. In other words, multi-block data structures have a hard time building on top of each other.
+
+Since there wasn’t a standardized way to describe these data structures we couldn’t build libraries for paths and selectors that seamlessly supported them.
+
+As we started designing this system several other requirements surfaced.
+
+* Transparent encryption envelopes on the read and write side.
+* Advanced `Link` types that can support some form of mutability and link to paths within other data structures.
```

Comment:
Not clear to me that this will end up being the same thing as the work we do to support mulitblock collections and transparent encryption.  I'd leave it out as a motivation here (although it surely is something that's been wished for).

Reply:
i want to capture it, even if we don’t have a great solution yet. it’s something we need to keep in mind as we desing the write interfaces.

Reply:
That's well and good, but capturing a list of things we want to think about, for which we don't have clear solutions we're about to write up yet... sounds like something that belongs in a design history notes document moreso than something we commit to in a specs-track document.

Reply:
Good point, I think I’ll break this into a section called “considerations” rather than “requirements.”